### DISCUSSION TITLE
Using DryIoc Container in a Xamarin Forms app in c# and I have problem with resolve of services

### DISCUSSION THREAD

**Discussion by nmpasdanis** [dadhi/DryIoc#538] [Q&A]
Currenly, I develop a Weather App in Xamarin Forms and I register the Services in App.xaml.cs by the help of the Instance of the Container Class. In the Container Class there is also a method the register the Services using also ( Reuse.Signleton ) and one other method that resolves the services. My question is that: I don't know if it is correctly using Reuse.Singleton because in reality I don't use it very much in this app. But if I remove the Reuse.Singleton from the Register method, my app will crash when as a user I will navigate from the one page to another, cause it will register and resolve again the same Services, please help me.Thank you very much community!!!

**Comment by dadhi** [comment]
Hi, I think you may get a faster help if you dub you question on StackOverflow, tagging with dryioc and xamarin.

**Reply by nmpasdanis** [reply]
Ok thank you man!!

**Comment by nmpasdanis** [comment]
Maksim you are the best.
It was a problem that you have resolved but I have not noticed it. Thank you very much!!
https://github.com/dadhi/DryIoc/issues/90

**Reply by dadhi** [reply]
Cheers.
Side question, are you developing for iOS?

**Reply by nmpasdanis** [reply]
Yes my friend and also for Android.

**Reply by dadhi** [reply]
Then, what method are you using to determine the platform is iOS?
There is a related issue I want to solve here #532.

**Comment by nmpasdanis** [comment]
By this one, I detect which platform my app will use in run time. I don't know if it 100% correct but it helped me to find a solution about my app's crashing.

            if (Xamarin.Forms.Device.RuntimePlatform == Xamarin.Forms.Device.iOS)
            {
                _container = new DryIoc.Container(rules: Rules.Default.WithUseInterpretation());
            }
            else
            {
                _container = new DryIoc.Container();
            }

**Reply by dadhi** [reply]
Thanks a lot.
